docs/architecture/notes.md

```diff
@@ -28,114 +28,114 @@ These components are:
 
 ### Assets
 
-> An [asset](assets.md) container for a note.
+> An [asset](assets.md) container for a `Note`.
 
-A note can contain up to `256` different assets. These assets represent fungible or non-fungible tokens, enabling flexible asset transfers.
+A `Note` can contain up to 256 different assets. These assets represent fungible or non-fungible tokens, enabling flexible asset transfers.
 
 ### Script
 
-> The code executed when the note is consumed.
+> The code executed when the `Note` is consumed.
 
-Each note has a script that defines the conditions under which it can be consumed. When accounts consume notes in transactions, note scripts call the account’s interface functions. This enables all sorts of operations beyond simple asset transfers. The Miden VM’s Turing completeness allows for arbitrary logic, making note scripts highly versatile.
+Each `Note` has a script that defines the conditions under which it can be consumed. When accounts consume `Note`s in transactions, `Note` scripts call the account’s interface functions. This enables all sorts of operations beyond simple asset transfers. The Miden VM’s Turing completeness allows for arbitrary logic, making `Note` scripts highly versatile.
 
 ### Inputs
 
-> Arguments passed to the note script during execution.
+> Arguments passed to the `Note` script during execution.
 
-A note can have up to `128` input values, which adds up to a maximum of 1 KB of data. The note script can access these inputs. They can convey arbitrary parameters for note consumption. 
+A `Note` can have up to 128 input values, which adds up to a maximum of 1 KB of data. The `Note` script can access these inputs. They can convey arbitrary parameters for `Note` consumption. 
 
 ### Serial number
 
-> A unique and immutable identifier for the note.
+> A unique and immutable identifier for the `Note`.
 
-The serial number helps prevent linkability between the note’s hash and its nullifier. It should be a random `Word` chosen by the user. If leaked, the note’s nullifier can be easily computed, potentially compromising privacy.
+The serial number helps prevent linkability between the `Note`’s hash and its nullifier. It should be random 32 bytes chosen by the user. If leaked, the `Note`’s nullifier can be easily computed, potentially compromising privacy.
```

Comment:
I think here, we need to say that the serial number has two goals: 

1. For private notes: Breaking linkability of note hash and note nullifier
2. For all notes: Ensuring uniqueness of the note. Using a random serial number makes the note unique. Using always the same serial number would create unspendable notes (since some would have the same nullifier, e.g. when sending the same asset to the same person using a P2ID note). The nullifier makes the difference